colour/colorimetry/spectrum.py

```diff
@@ -3690,9 +3939,10 @@ def interpolate(self, shape=SpectralShape(), method=None):
 
     def align(self,
               shape,
-              method='Constant',
-              left=None,
-              right=None):
+              interpolation_method=None,
+              extrapolation_method='Constant',
+              extrapolation_left=None,
```

Comment:
Hm...it's a weird interaction between the parameters. Wondering if this could be more elegant another way. Maybe allow passing an actual function for 'extrapolation_method'? But that might not work for `Linear`